### PR TITLE
evcxr: init at 0.3.3

### PR DESCRIPTION
**Pull request by protoben**
###### Motivation for this change

[evcxr](https://github.com/google/evcxr) is a Rust REPL. It's lacking some features, but it's relatively polished, and it's actively maintained. I'd like to see it in nixpkgs. 

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/interpreters/evcxr/default.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    repo = "evcxr";
+    rev = "ae07ccf08723b7aec0de57d540822b89088ca036";
+    sha256 = "1apc93z9vvf6qks5x2pad45rnrj9kjl812rj78w5zmmizccp2fhf";
+  };
+
+  cargoSha256 = "1pvw1jfi4p0cla4xxzgxh1jskvbmsyi4l5cimiis7093hyjs0svw";
+  cargoPatches = [ ./cargo-lock.patch ];
+
+  nativeBuildInputs = [ pkgconfig zeromq ];
+  buildInputs = [ cargo makeWrapper ];
```
Comment (symphorien) [inline pkgs/development/interpreters/evcxr/default.nix]: makeWrapper should be a `nativeBuildInput`.

File: pkgs/development/interpreters/evcxr/default.nix [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    owner = "google";
+    repo = "evcxr";
+    rev = "ae07ccf08723b7aec0de57d540822b89088ca036";
+    sha256 = "1apc93z9vvf6qks5x2pad45rnrj9kjl812rj78w5zmmizccp2fhf";
+  };
+
+  cargoSha256 = "1pvw1jfi4p0cla4xxzgxh1jskvbmsyi4l5cimiis7093hyjs0svw";
+  cargoPatches = [ ./cargo-lock.patch ];
+
+  nativeBuildInputs = [ pkgconfig zeromq ];
```
Comment (symphorien) [inline pkgs/development/interpreters/evcxr/default.nix]: zeromq is likely to be a buildInput.

File: pkgs/development/interpreters/evcxr/default.nix (hunk 1 above)
Comment (protoben) [reply pkgs/development/interpreters/evcxr/default.nix]: Indeed. Fixed.

File: pkgs/development/interpreters/evcxr/default.nix (hunk 2 above)
Comment (protoben) [reply pkgs/development/interpreters/evcxr/default.nix]: zeromq was only needed by the Jupyter kernel. I figure that should probably be its own package, so I've removed it from the build, which eliminates the dependency.

**Comment by Ma27** [comment]
@GrahamcOfBorg build evcxr

**Comment by Ma27** [comment]
Fixed the `cargoSha256` hash that needed to be altered after the last `cargo-vendor` bump. Also ensured that `cc` is always available by wrapping the binary.

@GrahamcOfBorg build evcxr

**Comment by Ma27** [comment]
@GrahamcOfBorg build evcxr

**Review by Ma27 — APPROVED** [review]

**Comment by Ma27** [comment]
Built and tested locally, appears to work fine, thanks a lot for working on this!

**Pull request merged**